src/java.base/share/classes/java/lang/AbstractStringBuilder.java

```diff
@@ -991,6 +1005,9 @@ public AbstractStringBuilder replace(int start, int end, String str) {
         shift(end, newCount - count);
         this.count = newCount;
         putStringAt(start, str);
+        if (end - start > 0) {
```

Comment:
regardless of value of `end - start` you could also skip setting `maybeLatin1 = true` if:
- `str.coder() == UTF16`
- `this.coder == LATIN1`

Reply:
hi, @cl4es, 
you are correct for this and the comment at setCharAt(), but I don't think it's necessary to check all cases. this attribute is just like a hint. if this.coder == LATIN1, that it doesn't matter if maybeLatin1 is true. 

if our attitude is checking all cases, it will become too complex and error-prone. deleteCharAt() and setLength() also need to check.  if so, it will pollute code more. I incline to set this attribute conservatively in all deleting methods.